packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -4415,10 +4415,18 @@ namespace Tpetra {
         auto lclRowMap = this->rowMap_->getLocalMap ();
         auto lclColMap = this->colMap_->getLocalMap ();
 
+        // FENCE REVIEW - NOW PASSING
+        //   Testing: This code is exercised by unit tests.
+        //   GTX960:  Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+        //   White:   Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+        //   Plan:    Remove this fence.
+        //   Notes:   determineLocalTriangularStructure appears to be properly
+        //            setup as device.
+
         // Make sure that the GPU can see any updates made on host.
         // This code only reads the local graph, so we don't need a
         // fence afterwards.
-        execution_space().fence ();
+        // execution_space().fence ();
```

Comment:
@kddevin In my opinion, we need this fence even though it passes tests. Map uses kokkos view with relying on UVM (it does not have a dual view context). So, the values can be touched somewhere and we need to make sure the updates should be done before we use it. 

Passing unit tests means that we so far do not have a case that other functions touch the UVM views in the map before this function is used. However, it does not mean that this use case is right.

Reply:
Updated comments #7546